g3doc/faq.md

```diff
@@ -0,0 +1,308 @@
+# Frequently Asked Questions
+
+[[_TOC_]]
+
+## Documentation
+
+1.  How do I **find the Highway op name** corresponding to existing intrinsics?
+    Search for the intrinsic in (for example)
+    [x86_128-inl.h](https://github.com/google/highway/blob/master/hwy/ops/x86_128-inl.h).
+    The Highway op is typically the name of the function that calls the
+    intrinsic. See also the
+    [quick reference](https://github.com/google/highway/blob/master/g3doc/quick_reference.md)
+    which lists all Highway ops.
+
+1.  Are there **examples of porting intrinsics to Highway**? See cl/448957386
+    and cl/450480902.
+
+1.  Where do I find documentation for each **platform's intrinsics**? See
+    [Intel](https://www.intel.com/content/www/us/en/docs/intrinsics-guide),
+    [Arm NEON and SVE](https://developer.arm.com/architectures/instruction-sets/intrinsics),
+    [RISC-V V](https://github.com/riscv/riscv-v-spec/blob/master/v-spec.adoc),
+    [WebAssembly](https://nemequ.github.io/waspr/intrinsics).
+
+1.  Where do I find **instruction latency/throughput**? For x86, a combination
+    of [uops.info](https://www.uops.info/table.html) and
+    https://agner.org/optimize/, plus Intel's above intrinsics guide and
+    [AMD's sheet (zip file)](https://www.amd.com/system/files/TechDocs/56665.zip).
+    For Arm, the Software_Optimization_Guide for Neoverse N1/V1 etc. For RISC-V,
+    the vendor's tables (typically not publicly available).
+
+1.  Where can I find **inspiration for SIMD-friendly algorithms**?
+
+    -   [Algorithms for Modern Hardware online book](https://en.algorithmica.org/hpc/)
+    -   [SIMD for C++ developers](http://const.me/articles/simd/simd.pdf)
+    -   [Bit twiddling collection](https://graphics.stanford.edu/~seander/bithacks.html)
+    -   [SIMD-within-a-register](http://aggregate.org/MAGIC/)
+    -   Hacker's Delight book (huge collection of bitwise identities, but
+        written for hypothetical RISC CPUs, which differ in some ways from the
+        SIMD capabilities of current CPUs).
+
+1.  How do I **predict performance**? The best approach by far is end-to-end
+    application benchmarking. Typical microbenchmarks are subject to numerous
+    pitfalls including unrealistic cache and branch predictor hit rates (unless
+    the benchmark randomizes its behavior). But sometimes we would like a quick
+    indication of whether a short piece of code runs efficiently on a given CPU.
+    Intel's IACA used to serve this purpose but has been discontinued. We now
+    recommend llvm-mca,
+    [integrated into Compiler Explorer](https://gcc.godbolt.org/z/n-KcQ-). This
+    shows the predicted throughput and the pressure on the various functional
+    units, but does not cover dynamic behavior including frontend and cache. For
```

Comment:
FWIW, uica does, but it only supports Intel CPUs (and doesn't support the newest/Golden Cove):
https://uica.uops.info/
https://arxiv.org/pdf/2107.14210.pdf

Reply:
Thanks, added :)